netket/utils/static_range.py

```diff
@@ -70,6 +74,8 @@ class StaticRange(struct.Pytree):
     """The first value in the range."""
     step: float = struct.field(pytree_node=False)
     """The difference between two consecutive values in the range."""
+    end: float = struct.field(pytree_node=False)
+    """The difference between two consecutive values in the range."""
```

Comment:
```suggestion
```

you can add a bit below a 
```python
@property
def end(self):
    return self.start+self.end*self.step
```

Reply:
Since `StaticRange` inherits from `PyTree`, shouldn't the attributes be immutable already?
I will change it to make it clearer.

Reply:
Yeah, in theory.
But the point is that I want to keep the minimum number of 'stored attributes' (start, step, length), and have all the other be properties that are computed from the stored attributes.

Maybe from an user point of view is the same, but from a conceptual point of view this is cleaner.
There's no need to store `end`, and if you store it, you now have the possibility of it getting out of sync from the length. 
the code is likely correct, but if you don't store end in the first place, you avoid the possibility for this kind of bugs.